### PR TITLE
Add support for duration primitive type in code generation

### PR DESCRIPTION
**Pull request by gtalha07**
- [x] Implement `render_literal` function for Duration literals.
- [x] Add `impl_code_type_for_primitive!` and `impl_renderable_for_primitive!` instances for `DurationCodeType`.
- [x] Add `liftDuration` and `lowerDuration` functions to `generate_wrapper_lifters` and `generate_wrapper_lowerers` respectively.
- [x] Update `generate_wrapper_lifters` and `generate_wrapper_lowerers` to handle Duration type.
- [x] Update tests.

### CODE REVIEW

**Review by gnunicorn — CHANGES_REQUESTED** [review]
looking good. But how did we land on adding tests for this?

**Comment by gtalha07** [comment]
> looking good. But how did we land on adding tests for this?

You mean these tests aren't necessary or enough for checking duration representation ? Should I remove this ?

**Review by gnunicorn — APPROVED** [review]
sorry, missed the test files somehow. all good.

File: fixtures/duration_type_test/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#[uniffi::export]
+pub fn get_back_duration_milliseconds(value: u64) -> u64 {
+    value
+}
+
+#[uniffi::export]
+pub fn get_back_duration_microseconds(value: u64) -> u64 {
+    value
+}
```
Comment (gnunicorn) [inline fixtures/duration_type_test/src/lib.rs:31]: these aren't testing the duration, this is testing `u64`...

**Review by gnunicorn — CHANGES_REQUESTED** [review]
ehem... better to actually read the code before approving ... the tests are not testing the duration type but u64 types...

File: fixtures/duration_type_test/test/duration_type_test.dart [hunk 2]
```diff
@@ -3,22 +3,7 @@ import '../duration_type_test.dart';
...
-  });
-  test('getBackDurationMilliseconds', () {
-    expect(api.getBackDurationMilliseconds(4000), 4000);
-  });
-  test('getBackDurationMicroseconds', () {
-    expect(api.getBackDurationMicroseconds(4000000), 4000000);
-  });
+  final duration = api.addDuration(1, 1);
+  expect(api.getSeconds(duration), equals(1));
+  expect(api.getNanos(duration), equals(1));
```
Comment (gnunicorn) [inline fixtures/duration_type_test/test/duration_type_test.dart:8]: maybe we can use some more complicated values than just a `1`?

Comment (gtalha07) [reply fixtures/duration_type_test/test/duration_type_test.dart:8]: Yeah I'll add more test cases, the PR is not yet finalised. I am beginning to see actual errors in type generator now...

File: src/gen/types.rs [hunk 3]
```diff
@@ -365,6 +365,7 @@ pub fn generate_type(ty: &Type) -> dart::Tokens {
         | Type::Int64
         | Type::UInt16
         | Type::Int32
+        | Type::Duration
```
Comment (gnunicorn) [inline src/gen/types.rs:368]: I am not sure this is needed at all....

File: src/gen/primitives.rs [hunk 4]
```diff
@@ -22,7 +24,8 @@ fn render_literal(literal: &Literal) -> String {
             | Type::UInt32
             | Type::UInt64
             | Type::Float32
-            | Type::Float64 => num_str,
+            | Type::Float64
+            | Type::Duration => num_str,
```
Comment (gnunicorn) [inline src/gen/primitives.rs:28]: neither that this is actually needed.

**Review by chavic — APPROVED** [review]
This looks good, and it looks like the tests passed

**Pull request merged**